test/api.test.js

```diff
@@ -0,0 +1,374 @@
+        )
+        todo--
+      })
+    )
+
+    pull(
+      sbot.metafeeds.branchStream({
+        tombstoned: false,
+        old: true,
+        live: false,
+      }),
+      pull.collect((err, branches) => {
+        if (err) throw err
+        t.equal(branches.length, 4, 'tombstoned: false')
+        todo--
+      })
+    )
+
+    pull(
+      sbot.metafeeds.branchStream({ tombstoned: null, old: true, live: false }),
+      pull.collect((err, branches) => {
+        if (err) throw err
+        t.equal(branches.length, 5, 'tombstoned: null')
+        todo--
+      })
+    )
+  }
+
+  await p(sbot.metafeeds.advanced.findAndTombstone)(
+    rootMF,
+    (f) => f.feedpurpose === 'chess',
+    'This game is too good'
+  )
+
+  while (todo > 0) {
+    // eslint-disable-line
+    await p(setTimeout)(100)
+  }
+
+  console.log('(persistence)')
+  await p(sbot.close)()
+  sbot = Testbot({ path, rimraf: false })
+
+  todo = 3
+  testOldStreams(sbot)
+  while (todo > 0) {
+    // eslint-disable-line
+    console.log('doop')
+    await p(setTimeout)(100)
+  }
+
+  sbot.close()
+  t.end()
+})
+
+test('findOrCreate() recps', async (t) => {
+  const sbot = Testbot()
+
+  const testkey = Buffer.from(
+    '30720d8f9cbf37f6d7062826f6decac93e308060a8aaaa77e6a4747f40ee1a76',
+    'hex'
+  )
+  sbot.box2.setOwnDMKey(testkey)
+
+  const mf = await p(sbot.metafeeds.advanced.findOrCreate)()
+  const f = await p(sbot.metafeeds.advanced.findOrCreate)(
+    mf,
+    (f) => f.feedpurpose === 'private',
+    {
+      feedpurpose: 'private',
+      feedformat: 'classic',
+      metadata: {
+        recps: [sbot.id],
+      },
+    }
+  )
+
+  t.equal(f.feedpurpose, 'private')
+  t.equal(f.metadata.recps[0], sbot.id)
+  sbot.close()
+  t.end()
+})
+
+test('findOrCreate (sugar!)', async (t) => {
+  const sbot = Testbot()
+
+  const details = {
+    feedpurpose: 'chess',
+    // feedformat: 'classic', optional
+  }
+  const chessMF = await p(sbot.metafeeds.findOrCreate)(details)
+  t.equal(chessMF.feedpurpose, details.feedpurpose, 'creates feed')
+
+  const chessMF2 = await p(sbot.metafeeds.findOrCreate)(details)
+  t.deepEqual(chessMF, chessMF2, 'finds feed')
+
+  await new Promise((resolve) => {
+    pull(
+      sbot.metafeeds.branchStream({ root: null, old: true, live: false }),
+      pull.collect((err, branches) => {
+        if (err) throw err
+        // console.log(branches.map(branch => branch.map(f => f[1] && f[1].feedpurpose)))
+
+        t.equal(branches.length, 5, 'correct number of feeds created')
+        // root, v1, shard, chess (AND MAIN)
+
+        const purposePath = branches.pop().map((f) => f[1] && f[1].feedpurpose)
+        t.deepEqual(purposePath, [null, 'v1', purposePath[2], 'chess'])
+        // TODO it would be nice for testing that we could deterministically know the shard
+        // but I don't know how to fix the "seed" that the root feed is derived from
+        resolve()
+      })
+    )
+  })
+
+  sbot.close()
+  t.end()
+})
```

Comment:
TODO - consider splitting out the "advanced" methods into own tests
Perhaps extract high level sugar methods into own tests - these are more integration tests, but nice to give them space